docs/book/book.typ

```diff
@@ -0,0 +1,178 @@
+
+
+- Maintain Existing Lucene Integration - Avoid changes to the Scala glue code that communicates with Lucene. This minimizes complexity and ensures that proven indexing and search logic remains intact.
+
+
+- Preserve On-Disk Data Format - Do not alter existing index files. This prevents re-indexing, which would otherwise introduce downtime and operational risk.
+
+
+- Deliver a Drop-In Replacement - The new version of Clouseau must be deployable as a drop-in replacement for the current system. This simplifies rollout and rollback strategies, making deployment safer and faster.
+
+
+- Use a Modern Foundation - Replace the unsupported components with a modern, well-supported technology stack. This improves maintainability, security, and long-term viability.
+
+
+- Retain Business Logic - Keep the existing business logic intact to reduce functional risk and ensure predictable behavior during migration.
+
+
+- Avoid Building Custom Infrastructure - Do not implement a custom asynchronous scheduler or other low-level components. Instead, leverage mature libraries and frameworks to reduce complexity and improve reliability.
+
+
+- Enable Future Enhancements - While the initial phase focuses on stability and compatibility, the new foundation should allow future improvements such as Lucene upgrades, API enhancements, and performance optimizations without requiring another full rewrite.
+
+#pagebreak()
+
+== Updated and migrated components
+
+#table(
+  columns: 4,
+  [*Dependency*], [*Old Version*], [*Replacement*], [*Replacement version*],
+  [Java], [JDK 7], [Java], [JDK 21.0.12],
+  [Scala], [2.9.1], [Scala], [2.13.16],
+  table.cell(rowspan: 2)[com.yammer.metrics], table.cell(rowspan: 2)[2.2.0],
+    [zio.metrics], [2.0.0],
+    [micrometer], [1.16.0],
+  [jetlang], [0.2.12], [zio], [2.1.16],
+  [scalang], [2.9.1], [-], [-],
+  table.cell(rowspan: 2)[org.slf4j], table.cell(rowspan: 2)[1.7.32],
+    [zio.logging], [2.5.0],
+    [tinylog], [2.7.0],
+  [commons-configuration], [1.8], [zio.config], [4.0.4],
+)
+
+= Architecture
+
+
+The project follows a layered architecture where all components are organized into distinct layers:
+
+- Foundation Layer: This layer includes core dependencies, such as ZIO and OTP abstractions, which form the foundation of our application. The motivation behind splitting the foundation into core and otp was to enable potential future replacements of communication protocol while retaining the existing functionality.
+
+- Facade Layer: This layer consists of `scalang` emulation and interfaces converters, which act as a facade for the application, allowing us to run unmodified buisness logic from previous version of the project.
+
+- Clouseau Layer: The main application logic resides in this layer, primarily made up of unmodified service classes like ClouseauSupervisor, IndexManagerService, IndexManager, AnalyzerService. These classes are responsible for implementing the core business functionality.
```

Comment:
Perhaps we should call this the "application layer"?  Clouseau itself should be composed of all the layers in this stack.